src/components/App/PuzzleGenerate/index.tsx

```diff
@@ -25,87 +26,100 @@ const PuzzleGenerate = ({ puzzlesList }) => {
         hint
       );
 
-      if (puzzleInstance.error) {
+      if (puzzleInstance?.error) {
         throw new Error(puzzleInstance.message);
-      } else {
-        setPuzzleInstanceData(puzzleInstance);
       }
     };
     toast.promise(puzzleInstancePromise(), {
       loading: 'Making your puzzle instance... ⚙️',
-      success: 'Success',
+      success: 'Successfully added new puzzle instance',
       error: err => err.message
     });
   };
 
   return (
     <>
       <div className={styles.form}>
-        <h2> Make a puzzle instance </h2>
-        <Input
-          type="text"
-          id="puzzleHint"
-          required={true}
-          placeholder="Hint"
-          setInputVal={setHint}
-        />
-        <div className={styles.puzzleLocation}>
-          <Input
-            type="text"
-            id="puzzleLatitude"
-            required={true}
-            placeholder="Latitude"
-            setInputVal={setLatitude}
-          />
-          <Input
-            type="text"
-            id="puzzleLongitude"
-            required={true}
-            placeholder="Longitude"
-            setInputVal={setLongitude}
-          />
-          <Input
-            type="text"
-            id="puzzleAddress"
-            required={true}
-            placeholder="Address"
-            setInputVal={setAddress}
-          />
-        </div>
-        <div>
-          {puzzlesList.length > 0 && (
+        <h3> Set New Puzzle Location </h3>
+        <div className={styles.selectionWrapper}>
+          {puzzlesList.length > 0 ? (
             <select
               className={styles.selections}
               value={puzzleId}
               onChange={e => setPuzzleId(e.currentTarget.value)}
             >
-              <option selected>Choose a puzzle</option>
-              {puzzlesList.map((puzzle, index) => (
+              <option>Choose a puzzle</option>
+              {puzzlesList.map((puzzle: PuzzleCustom, index: number) => (
                 <option value={puzzle.id} key={`puzzle${index}`}>
                   {puzzle.name}
                 </option>
               ))}
             </select>
+          ) : (
+            <>
+              <select className={styles.selections} value="No puzzle available">
+                \
+              </select>
+            </>
           )}
         </div>
+        <div className={styles.hintWrapper}>
+          <Input
+            type="text"
+            id="puzzleHint"
+            required={true}
+            placeholder="Hint"
+            setInputVal={setHint}
+          />
+        </div>
+        <div className={styles.addressWrapper}>
+          <Input
+            id="address-displayer"
```

Comment:
Can we use snek case for ids?

Reply:
we can! i thought js prefer `addressDisplayer`